proposals/proposal-0001.md

```diff
@@ -0,0 +1,221 @@
+---
+proposal: 0001
+title: "Writing a Proposal"
+authors: [ "Matthew Fisher <matt.fisher@microsoft.com" ]
+created: 2020-08-13
+---
+
+A proposal is a design document providing information to the Helm community, or
+describing a new feature for a Helm project or its processes or environment. The
+proposal should provide a concise technical specification of the feature and a
+rationale for the feature.
+
+We intend proposals to be the primary mechanisms for proposing major new
+features, for collecting community input on an issue, and for documenting the
+design decisions that have gone into the project. The proposal author is
+responsible for building consensus within the community and documenting
+dissenting opinions.
+
+The system is heavily influenced by Python's Enhancement Proposal (PEP) system,
+hence the name. Given that Helm is a much (MUCH) smaller project than Python,
+there are a few changes in the proposal workflow to make it a looser format for
+proposals, most notably the lack of proposal "editors", a BDFL and a much
+smaller review & resolution cycle for proposals.
+
+Because the proposals are maintained as markdown files in a versioned
+repository, their revision history is the historical record of the proposal.
+
+## Proposal workflow
+
+### Start with an idea for Helm
+
+The proposal process begins with a new idea for the Helm project. It is highly
+recommended that a single proposal contain a single key proposal or new idea.
+Small enhancements or patches often don't need a proposal and can be injected
+into a project's development workflow with a patch submission to the project's
+issue tracker. The more focused the proposal, the more successful it tends to
+be. As per Helm's governance, the project maintainers reserve the right to
+reject proposal proposals if they appear too unfocused, too broad, or
+incomplete. If in doubt, split your proposal into several well-focused ones, or
+explain your proposal more clearly.
+
+Each proposal must have a champion -- someone who writes the proposal using the
+style and format described below, shepherds the discussions in the appropriate
+communication channels, and attempts to build community consensus around the
+idea. The proposal champion (a.k.a. author) should first attempt to ascertain
+whether the idea is proposal-able. Posting to the helm-users mailing list is the
+best way to go about this.
+
+Vetting an idea publicly before going as far as writing a proposal is meant to
+save the potential author time. Many ideas have been brought forward for
+changing Helm that have been rejected for various reasons. Asking the Helm
+community first if an idea is original helps prevent too much time being spent
+on something that is guaranteed to be rejected based on prior discussions
+(searching the internet does not always do the trick). It also helps to make
+sure the idea is applicable to the entire community and not just the author.
```

Comment:
Seeing as this is one of the largest problems with Helm PRs right now, perhaps it needs to be bolded or made its own section so as to be very clear